### PR TITLE
Replace `TypePointer` with `Type const*`

### PR DESCRIPTION
**Pull request by Marenz**

### CODE REVIEW

File: libsolidity/analysis/TypeChecker.cpp [hunk 1]
```diff
@@ -76,13 +76,13 @@ bool TypeChecker::checkTypeRequirements(SourceUnit const& _source)
 	return Error::containsOnlyWarnings(m_errorReporter.errors());
 }
 
-TypePointer const& TypeChecker::type(Expression const& _expression) const
+Type const* const& TypeChecker::type(Expression const& _expression) const
```
Comment (chriseth) [inline libsolidity/analysis/TypeChecker.cpp]: Can you check these cases, please? I think this should return `Type const*`.

Comment (Marenz) [reply libsolidity/analysis/TypeChecker.cpp]: Yes, good point. I fixed the `const* const&` I saw, but the return value ones escaped my notice.
Now updated.

**Comment by axic** [comment]
Will this also drop the type definition for `TypePointer`?

**Comment by Marenz** [comment]
Yes, all type definitions were removed (I think I found about 3 or 4)

File: test/libsolidity/AnalysisFramework.h [hunk 2]
```diff
@@ -33,7 +33,6 @@ namespace solidity::frontend
 {
 class Type;
 class FunctionType;
-using TypePointer = Type const*;
 using FunctionTypePointer = FunctionType const*;
```
Comment (axic) [inline test/libsolidity/AnalysisFramework.h:36]: What about this?

Comment (axic) [reply test/libsolidity/AnalysisFramework.h:36]: Well, this is in the tests, so not too important.

File: libsolidity/formal/SymbolicVariables.cpp [hunk 3]
```diff
@@ -61,7 +61,7 @@ SymbolicVariable::SymbolicVariable(
 	solAssert(m_sort, "");
 }
 
-smtutil::Expression SymbolicVariable::currentValue(frontend::TypePointer const&) const
```
Comment (axic) [inline libsolidity/formal/SymbolicVariables.cpp:64]: @leonardoalt what was the intent here?

File: test/libsolidity/AnalysisFramework.h (hunk 2 above)
Comment (Marenz) [reply test/libsolidity/AnalysisFramework.h:36]: It's not the `TypePointer` and the request was explicitly for that. I can replace `FunctionTypePointer` too if that is desired.
@chriseth also wanted to keep `TypePointers`.

**Review by chriseth — APPROVED** [review]

**Pull request merged**